src/ui/map.js

```diff
@@ -1067,26 +1066,32 @@ class Map extends Camera {
     }
 
     /**
-     * Sets the map's language.
+     * Sets the map's language, which is used for translating map labels and UI components.
      *
-     * @private
-     * @param {string} language A string representing the desired language. `undefined` or `null` will remove the current map language and reset the map to the default language as determined by `window.navigator.language`.
+     * @param {'auto' | string} [language] A string representing the desired language used for the map's labels and UI components. Languages can only be set on Mapbox vector tile sources.
+     *  Valid language strings must be a [BCP-47 language code](https://en.wikipedia.org/wiki/IETF_language_tag#List_of_subtags). Unsupported BCP-47 codes will not include any translations. Invalid codes will result in an recoverable error.
+     *  If a label has no translation for the selected language, it will display in the label's local language.
+     *  If param is set to `auto`, GL JS will select a user's preferred language as determined by the browser's [`window.navigator.language`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/language) property.
+     *  If the `locale` property is not set separately, this language will also be used to localize the UI for supported languages.
+     *  If param is set to `undefined` or `null`, it will remove the current map language and reset the language used for translating map labels and UI components.
      * @returns {Map} Returns itself to allow for method chaining.
      * @example
      * map.setLanguage('es');
      *
      * @example
      * map.setLanguage('auto');
+     *
+     * @example
+     * map.setLanguage();
      */
-    setLanguage(language?: ?string): this {
-        this._language = language === 'auto' ? window.navigator.language : language;
+    setLanguage(language?: 'auto' | ?string): this {
+        const newLanguage = language === 'auto' ? window.navigator.language : language;
+        if (!this.style || newLanguage === this._language) return this;
+        this._language = newLanguage;
 
-        if (this.style) {
-            for (const id in this.style._sourceCaches) {
-                const source = this.style._sourceCaches[id]._source;
-                if (source._setLanguage) {
-                    source._setLanguage(this._language);
-                }
+        for (const source of this.style._getSources()) {
+            if (source.reload) {
+                source.reload();
```

Comment:
Nit: since this pattern is repeated a few times, should we define `style._reloadSources()` instead of `style._getSources` and use that?

Reply:
Good catch, thanks!